### PR TITLE
 app-office/moneydance: New package

### PR DESCRIPTION
**Pull request by ConiKost**
Moneydance is easy to use personal finance software
that is loaded with all the features you need:
online banking and bill payment, account management,
budgeting and investment tracking.

Closes: https://bugs.gentoo.org/19910
Signed-off-by: Conrad Kostecki <conrad@kostecki.com>
Package-Manager: Portage-2.3.51, Repoman-2.3.11

### CODE REVIEW

**Comment by ConiKost** [comment]
@gentoo/licenses: Could you have a look, if that's okay with those new licenses?
@chewi: Could you do a review, since it's java stuff?

File: app-office/moneydance/moneydance-2017.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+MY_PN="Moneydance"
+
+inherit desktop java-pkg-2 xdg-utils
+
+DESCRIPTION="A personal finance app for Mac, Windows, Linux, iPhone and iPad."
```
Comment (juippis) [inline app-office/moneydance/moneydance-2017.8.ebuild]: No dot please.

File: app-office/moneydance/moneydance-2017.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+pkg_nofetch() {
+	elog "Please download ${A} from"
+	elog "https://www.infinitekind.com/download-moneydance-personal-finance-software"
+	elog "and place it in your DISTDIR directory."
+}
+
+src_prepare() {
+	# Modify .desktop file, to fix QA errors
+	sed -e 's/Application;//g' -e 's/.png//g' -i resources/moneydance.desktop
```
Comment (juippis) [inline app-office/moneydance/moneydance-2017.8.ebuild]: `|| die`

File: app-office/moneydance/moneydance-2017.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	default
```
Comment (juippis) [inline app-office/moneydance/moneydance-2017.8.ebuild]: (nitpick, but) this should probably be the first thing to call in src_prepare.

Comment (ConiKost) [reply app-office/moneydance/moneydance-2017.8.ebuild]: Thanks, changed.

File: app-office/moneydance/moneydance-2017.8.ebuild (hunk 2 above)
Comment (ConiKost) [reply app-office/moneydance/moneydance-2017.8.ebuild]: Thanks, added.

File: app-office/moneydance/moneydance-2017.8.ebuild (hunk 1 above)
Comment (ConiKost) [reply app-office/moneydance/moneydance-2017.8.ebuild]: Thanks, removed.

**Comment by monsieurp** [comment]
@juippis: I can also take a look since I'm part of the Java team :P

File: app-office/moneydance/moneydance-2017.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+HOMEPAGE="https://www.moneydance.com/"
+SRC_URI="
+	amd64? ( https://infinitekind.com/stabledl/current/${MY_PN}_linux_amd64.tar.gz -> ${P}-amd64.tar.gz )
+	x86? ( https://infinitekind.com/stabledl/current/${MY_PN}_linux_x86.tar.gz -> ${P}-x86.tar.gz )
+"
+
+LICENSE="Apache-1.0 Apache-2.0 BSD CPAL-1.0 CPL-1.0 CSL-2.0 LGPL-2 MIT TIK"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
```
Comment (monsieurp) [inline app-office/moneydance/moneydance-2017.8.ebuild]: Make sure to add `DEPEND=>=virtual/jdk-1.8` as well.

**Comment by chewi** [comment]
Sorry, I did say I would look at this.

File: app-office/moneydance/moneydance-2017.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	java-pkg_dojar jars/*.jar
+	java-pkg_dolauncher moneydance --main "Moneydance" --java_args "-client -Dawt.useSystemAAFontSettings=gasp -Dawt.useSystemAAFontSettings=on -Xmx1024m"
+
+	doicon resources/*.png
+	domenu resources/moneydance.desktop
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+	xdg_mimeinfo_database_update
```
[the remark this answers is not in the document]
Comment (ConiKost) [reply app-office/moneydance/moneydance-2017.8.ebuild]: Thanks, you are right, no MimeType entry, updated :)

File: app-office/moneydance/moneydance-2017.8.ebuild (hunk 4 above)
Comment (ConiKost) [reply app-office/moneydance/moneydance-2017.8.ebuild]: Thanks, added :)

**Comment by monsieurp** [comment]
@chewi: is it good to go?

**Pull request closed without merging**